tests/rbd/rbd_utils.py

```diff
@@ -1158,3 +1160,150 @@ def verify_image_size(rbd, pool, image, size):
         return 0
     log.error(f"{image} size is not equal to expected size as {size}")
     return 1
+
+
+def resize_parent_and_clone(rbd, resize_sequence, parent_clone_spec):
+    """
+    Resize parent and clone images to compensate overhead due to the header size.
+
+    Scenarios and corresponding steps
+    1) Parent and clone are luks1 encrypted
+        Resize parent at beginning to compensate for header size,
+        clone resize not required
+    2) Parent luks1/luks2 and clone is not encrypted
+        Resize parent at beginning to compensate for header size,
+        clone resize not required since it has no header
+    3) Parent luks2 and clone luks1
+        Resize parent at beginning to compensate for header size,
+        resize clone with --allow-shrink since luks2 header is bigger than luks1
+    4) Parent not encrypted and clone is luks1/luks2
+        Resize not required before clone since parent is not encrypted, resize
+        clone with --allow-shrink post cloning to compensate for header
+    5) Parent luks1 and clone luks2
+        Resize parent before clone, resize both parent and
+        clone with --allow-shrink post cloning
+
+    Justification:
+    Since LUKS2 header is usually bigger than LUKS1 header, the rbd resize command at
+    the beginning temporarily grows the parent image to reserve some extra space in the
+    parent snapshot and consequently the cloned image. This is necessary to make all parent
+    data accessible in the cloned image. The rbd resize command at the end shrinks the
+    parent image back to its original size and does not impact the parent snapshot and
+    the cloned image to get rid of the unused reserved space
```

Comment:
[Optional]

1. Should we be moving some wrappers out of rbd_utils ? (may be something like rbd_encryption_utils) for this method to keep rbd_utils more maintainable ?
2. [minor] Justification part could be code comment or some link large method docs are okay ?

Reply:
1. We could do it, but for now I would like to keep it here, in my other draft PR for tier-3 scenarios I am already working on splitting relevant workflows into different python files, we could probably take this optimization up as part of that.
2. Is there any problem with large method docs? I am not aware, let me know if I need to remove justification I can.